scripts/Modules/mkw_classes/common.py

```diff
@@ -0,0 +1,301 @@
+from dolphin import memory, utils
+
+from dataclasses import dataclass
+from enum import Enum
+import math
+import struct
+
+class RegionError(Exception):
+    def __init__(self, message=f"Expected Mario Kart Wii game ID (RMCX01), " \
+                                 f"got {utils.get_game_id()}"):
+        super().__init__(message)
+
+@dataclass
+class vec2:
+    x: float = 0.0
+    y: float = 0.0
+
+    def __add__(self, other):
+        return vec2(self.x + other.x, self.y + other.y)
+
+    def __sub__(self, other):
+        return vec2(self.x - other.x, self.y - other.y)
+
+    @staticmethod
+    def read(ptr) -> "vec2":
+        bytes = memory.read_bytes(ptr, 0x8)
+        return vec2(*struct.unpack('>' + 'f'*2, bytes))
+
+@dataclass
+class vec3:
+    x: float = 0.0
+    y: float = 0.0
+    z: float = 0.0
+
+    def __add__(self, other):
+        return vec3(self.x + other.x, self.y + other.y, self.z + other.z)
+
+    def __sub__(self, other):
+        return vec3(self.x - other.x, self.y - other.y, self.z - other.z)
+
+    def norm_xyz(self) -> float:
+        return math.sqrt(self.x**2 + self.y**2 + self.z**2)
+
+    def norm_xz(self) -> float:
+        return math.sqrt(self.x**2 + self.z**2)
```

Comment:
```suggestion
    def length(self) -> float:
        return math.sqrt(self.x**2 + self.y**2 + self.z**2)

    def length_xz(self) -> float:
        return math.sqrt(self.x**2 + self.z**2)
```
`length` is a known `EGG::Vector3f` symbol.